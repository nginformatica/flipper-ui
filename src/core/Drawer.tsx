import MuiDrawer, { DrawerProps as MuiDrawerProps } from '@material-ui/core/Drawer'
import React, { FC } from 'react'
import { DefaultProps } from './types'
import { makeStyles, createStyles } from '@material-ui/core/styles'

interface DrawerProps extends MuiDrawerProps, DefaultProps {
    top?: number
    width?: number
}

const getStyles = (top?: number, width?: number) =>
    makeStyles(() =>
        createStyles({
            drawer: {
                top,
                width
            },
            drawerPaper: {
                top: 'inherit',
                width: 'inherit',
                height: top ? `calc(100% - ${top}px)` : '100%'
            }
        })
    )

const Drawer: FC<DrawerProps> = props => {
    const { style, margin, padding, top = 0, width, ...otherProps } = props
    const useStyles = getStyles(top, width)
    const classes = useStyles()

    return (
        <MuiDrawer
            { ...otherProps }
            classes={ {
                root: classes.drawer,
                paper: classes.drawerPaper
            } }
            style={ { margin, padding, ...style } }>
            { props.children }
        </MuiDrawer>
    )
}

export default Drawer
