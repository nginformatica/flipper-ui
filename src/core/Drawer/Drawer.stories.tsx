import React, { useState } from 'react'
import { Meta } from '@storybook/react'
import Drawer from '.'
import List from '../List'
import ListItem from '../ListItem'
import { Backup as IconBackup } from '../../icons'
import Button from '../Button'

export default {
    title: 'Drawer',
    component: Drawer
} as Meta<typeof Drawer>

export const Default = () => {
    const [open, SetOpen] = useState(false)

    const handleClick = () => {
        SetOpen(!open)
    }

    return (
        <>
            <Button onClick={handleClick} disabled={open}>
                Open
            </Button>
            <Drawer open={open} onClick={handleClick}>
                <List>
                    <ListItem icon={<IconBackup />} />
                    <ListItem icon={<IconBackup />} />
                </List>
            </Drawer>
        </>
    )
}
