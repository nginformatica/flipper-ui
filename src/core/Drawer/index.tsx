import MuiDrawer, {
    DrawerProps as MuiDrawerProps
} from '@material-ui/core/Drawer'
import React from 'react'
import { DefaultProps } from '../types'
import { makeStyles, createStyles } from '@material-ui/core/styles'

export interface DrawerProps extends MuiDrawerProps, DefaultProps {
    top?: number
    width?: number
}

const getStyles = (top: number, width?: number) =>
    makeStyles(() =>
        createStyles({
            drawer: {
                top,
                width
            },
            drawerPaper: {
                top: 'inherit',
                width: 'inherit',
                height: `calc(100% - ${top}px)`
            }
        })
    )

const Drawer = (props: DrawerProps) => {
    const { style, margin, padding, top = 0, width, ...otherProps } = props
    const useStyles = getStyles(top, width)
    const classes = useStyles()

    return (
        <MuiDrawer
            {...otherProps}
            classes={{
                root: classes.drawer,
                paper: classes.drawerPaper
            }}
            style={{ margin, padding, ...style }}>
            {props.children}
        </MuiDrawer>
    )
}

export default Drawer
