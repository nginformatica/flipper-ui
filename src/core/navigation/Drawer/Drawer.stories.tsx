import React, { useState } from 'react'
import { Meta } from '@storybook/react'
import Drawer from '.'
import List from '@/core/data-display/List'
import ListItem from '@/core/data-display/ListItem'
import { Backup as IconBackup } from '@/icons'
import Button from '@/core/inputs/Button'

export default {
    title: 'Navigation/Drawer',
    component: Drawer
} as Meta<typeof Drawer>

export const Default = () => {
    const [open, SetOpen] = useState(false)

    const handleClick = () => {
        SetOpen(!open)
    }

    return (
        <>
            <Button onClick={handleClick} disabled={open}>
                Open
            </Button>
            <Drawer open={open} onClick={handleClick}>
                <List>
                    <ListItem icon={<IconBackup />} />
                    <ListItem icon={<IconBackup />} />
                </List>
            </Drawer>
        </>
    )
}
