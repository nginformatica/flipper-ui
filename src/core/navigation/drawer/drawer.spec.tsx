import React from 'react'
import { render, screen } from '@testing-library/react'
import Drawer from '.'

describe('Drawer', () => {
    it('should render', () => {
        render(
            <Drawer open data-testid='drawer'>
                <div>Drawer</div>
            </Drawer>
        )

        const drawer = screen.getByTestId('drawer')
        const drawerContent = screen.getByText('Drawer')

        expect(drawer).toBeDefined()
        expect(drawerContent).toBeDefined()
    })

    it('should render nothing when closed', () => {
        const container = render(
            <Drawer open={false} data-testid='drawer'>
                <div>Drawer</div>
            </Drawer>
        )

        expect(container.container.firstChild).toBeNull()
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Drawer open data-testid='drawer'>
                <div>Drawer</div>
            </Drawer>
        )

        expect(container).toMatchSnapshot()
    })
})
