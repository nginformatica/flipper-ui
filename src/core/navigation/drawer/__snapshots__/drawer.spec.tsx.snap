// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should match snapshot 1`] = `
<div
  aria-hidden="true"
/>
`;
